### ISSUE TITLE
Update documentation for `add_weighted_edges_from`

### ISSUE THREAD

**Issue by joelmiller** [networkx/networkx#1974]
For `add_weighted_edges_from(ebunch, weight='weight', **attr)` the documentation gives

> **ebunch** (container of edges) – Each edge given in the list or container will be added to the graph. The edges must be given as 3-tuples (u,v,w) where w is a number.

There's no clear reason why `w` should be a number.  I looked at the source code and it didn't seem to require it.  So I tested it with `w` a string, and it worked.

**Comment by joelmiller** [comment]
And actually - while looking at this, there's another issue for me: it would be nice if `add_weighted_edges_from` would accept an input that looks just like the output of `G.edges(data=True)`.  This looks like a somewhat involved change.

**Comment by dschult** [comment]
To use the output of ```G.edges(data=True)``` you should just use ```H.add_edges_from()```.
The method ```add_weighted_edges_from()``` was created for the case when you have numeric data coming in that isn't in a dict form yet. 

    [(u, v, {'wt':2.3})]   -> add_edges_from
    [(u, v, 2.3)]     -> add_weighted_edges_from

It is true that you could use this to add nonnumeric data, but it was originally intended for numeric data.

**Comment by joelmiller** [comment]
hmm - I could have sworn I checked that `add_edges_from()` wouldn't accept this sort of input [that's why I looked for `add_weighted_edges_from()` in the first place].  Thanks for pointing out my mistake.  It's low priority, but I think it would be useful to update the documentation on `add_weighted_edges_from()`.  Something like "the edges must be given as 3-tuples `(u,v,w)` where `w` is the attribute of the u--v edge."   In particular I'm sure there are many cases where people are adding colored edges, but other contexts would also use non-numeric values.
